keps/sig-apps/4368-support-managed-by-label-for-batch-jobs/README.md

```diff
@@ -0,0 +1,985 @@
+for the EndpointSlices, and IPAddresses, respectively.
+
+#### Graduating directly to Beta
+
+The implementation is simple (see [Design details](#design-details)), short-circuit the
+`syncJob` invocation. Also, since the feature is just label-based, there is
+technically no need to go via Alpha in order to ensure that all
+kube-apiserver instances (in HA setup, see [here](https://kubernetes.io/releases/version-skew-policy/#supported-version-skew))
+recognize the new API, as in case of new fields. A similar approach was used
+to graduate the
+[Pod Index Label for StatefulSets and Indexed Jobs](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/4017-pod-index-label)
+feature with similar characteristics.
+
+Finally, we would like to have the feature soak without unnecessary delay.
+
+#### Can the label be mutable?
+
+There is a potential risk of leaking pods, if the value of the label is changed.
+For example, assume there is a running Job, which is reconciled by the Job
+controller, and has some pods created.
+Then, if the label is switched to the mirroring Kueue controller (which by
+itself does not manage pods). Then, the pods are leaking and remain running.
+
+In order to avoid the risk of pods leaking between the controllers when changing
+value of the `managed-by` label, we make the label immutable (see also
+[label mutability](#label-mutability)).
+
+However, the question remains if we can make the label mutable when the job is
+stopped, similarly, as we do with the `AllowMutableSchedulingDirectives` flag
+which guards mutability of the Job's pod template labels.
+
+It seems possible, and could be handy in [Story 1](#story-1), but it is also not
+a blocker.
+
+It would also the complicate debuggability of the feature.
+
+We decide to keep the label immutable, at least for [Beta](#beta), we will
+re-evaluate the decision for [GA](#ga).
+
+### Risks and Mitigations
+
+#### Ecosystem fragmentation due to forks
+
+The mechanism to disable the main Job controller opens the door for users to
+substitute it with a fork. This may create more fragmentation in the community
+as users may prefer to use their forked controllers rather than contribute
+upstream.
+
+First, this risk, to some extent, exists even today as admins with access to the
+control plane can disable by passing `--controllers=-job,*` in the manifest for
+`kube-controller-manager` (see more info on the `--controllers` flag
+[here](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/)).
+
+Second, we believe that users who had the need to fork the Job controller
+already introduced dedicated Job CRDs for their needs.
+
+#### Two controllers running at the same time on old version
+
+It is possible that one configures MultiKueue (or another project) with an
+older version of k8s which does not support the label yet. In that case
+two controllers might start running and compete with Job status updates at the
```

Comment:
It might be worth adding that this can also happen when a feature is disabled to this section.

Reply:
Sure, added a note about that scenario

Reply:
thx